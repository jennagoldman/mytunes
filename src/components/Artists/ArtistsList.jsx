import React from 'react';
import { Link } from 'react-router-dom';
import ArtistItem from './ArtistItem.jsx';
import PropTypes from 'prop-types';

const ArtistsList = ({ artists }) => {
  const artistsListItems = artists.map(artist => (
    <li key={artist.artistId}>
      <Link to={`/artist/${artist.artistId}`}>
        <ArtistItem {...artist} />
      </Link>
    </li>
  ));

  return (
    <section>
      <ul>
        {artistsListItems}
      </ul>
    </section>
  );
};

ArtistsList.propTypes = {
  artists: PropTypes.arrayOf(PropTypes.shape({
    artistId: PropTypes.string.isRequired,
    artistName: PropTypes.string.isRequired
  })).isRequired
};

export default ArtistsList;
