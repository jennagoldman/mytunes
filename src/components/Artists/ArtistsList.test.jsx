import React from 'react';
import { shallow } from 'enzyme';
import ArtistsList from './ArtistsList.jsx';

describe('ArtistsList', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<ArtistsList artists={[{ artistName: 'Electric Light Orchestra' }, { artistName: 'Pretty Lights' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
