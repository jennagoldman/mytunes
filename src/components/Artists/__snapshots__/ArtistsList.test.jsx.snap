// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsList matches a snapshot 1`] = `
<section>
  <ul>
    <li>
      <Link
        to="/artist/undefined"
      >
        <ArtistItem
          artistName="Electric Light Orchestra"
        />
      </Link>
    </li>
    <li>
      <Link
        to="/artist/undefined"
      >
        <ArtistItem
          artistName="Pretty Lights"
        />
      </Link>
    </li>
  </ul>
</section>
`;
